File: samples/FsBulletML.Sample.MonoGame.FSharp/Fps.fs
namespace FsBulletML.Sample.MonoGame.FSharp

open System
open Microsoft.Xna.Framework
open Microsoft.Xna.Framework.Graphics

type Fps () as this =
  [<DefaultValue>]val mutable value : float32
  [<DefaultValue>]val mutable private frameCount : float32
  [<DefaultValue>]val mutable private updateTimer : float32
  [<DefaultValue>]val mutable private interval : float32

  do
    this.interval <- 1.f

  member this.Update(delta:float) =
    this.frameCount <- this.frameCount + 1.f
    this.updateTimer <- this.updateTimer + float32 delta
    if (this.updateTimer > this.interval) then
      this.value <- this.frameCount / this.updateTimer
      this.frameCount <- 0.f
      this.updateTimer <- this.updateTimer - this.interval

File: samples/FsBulletML.Sample.MonoGame.CSharp/Fps.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace FsBulletML.Sample.MonoGame.CSharp
{
    class Fps
    {
        public float Value {get; private set;}
        private float interval;
        private float updateTimer;
        private int frameCount;

        public Fps()
        {
            this.Value = 0.0f;
            this.interval = 1.0f;
            this.updateTimer = 0.0f;
            this.frameCount = 0;
        }

        public void Update(float delta)
        {
            this.frameCount++;

            this.updateTimer += delta;

            if (this.updateTimer > this.interval)
            {
                this.Value = this.frameCount / this.updateTimer;
                this.frameCount = 0;
                this.updateTimer -= this.interval;
            }
        }
    }
}
